### ISSUE TITLE
Cannot compile on Debian Jessie

### ISSUE THREAD

**Issue by seemoss** [kozec/dumbxinputemu#3]
Hi,

I cannot compile dumbxinputemu 0.3.1 on Debian Jessie due to an unknown type (see below, same for 64-bit). I must be missing some dev library/include package but can't tell which one.

Regards,

Simos

```

$ make
/usr/bin/i686-w64-mingw32-gcc  -I/usr/i686-w64-mingw32/include -Idumbxinputemu -L{USR}/i686-w64-mingw32/lib/ -c dumbxinputemu/dinput_input.c -o dumbxinputemu/dinput_input.32.o
/usr/bin/i686-w64-mingw32-gcc  -I/usr/i686-w64-mingw32/include -Idumbxinputemu -L{USR}/i686-w64-mingw32/lib/ -c xinput1_3/xinput1_3.c -o xinput1_3/xinput1_3.32.o
xinput1_3/xinput1_3.c:51:5: error: unknown type name ‘_Inout_’
     _Inout_  XINPUT_VIBRATION *pVibration
     ^
Makefile:28: recipe for target 'xinput1_3/xinput1_3.32.o' failed
make: *** [xinput1_3/xinput1_3.32.o] Error 1

```

**Comment by kozec** [comment]
I'm not sure why should that one be undefined, but can you check output of `/usr/bin/i686-w64-mingw32-gcc --version` ?

**Comment by seemoss** [comment]
Thanks for the quick response. Here's what I have:

$ /usr/bin/i686-w64-mingw32-gcc --version
i686-w64-mingw32-gcc (GCC) 4.9.1

The dumbxinputemu-sal-includes branch gives the same error.

I'll be upgrading soon to Debian Stretch which was released a couple of months ago and seems to include mingw-w64 version 6.3.0 so don't waste any more time on this until I upgrade & test.

**Comment by kozec** [comment]
> i686-w64-mingw32-gcc (GCC) 4.9.1

Woa. I believe that 7.1.1 is _current_ version of gcc.

I'd suggest you to use precompiled dlls from Releases section in this specific case.

**Comment by C0rn3j** [comment]
✘ c0rn3j@c0rn3jDesktop  ~  /usr/bin/i686-w64-mingw32-gcc --version
i686-w64-mingw32-gcc (GCC) 7.1.1 20170516

Yup. 

If [this](https://sourceforge.net/p/mingw-w64/mailman/message/34269600/) is the actual project, then that type name was added 2015-07-06, and if the version numbers follow gcc - [which they seem to](https://archive.archlinux.org/packages/m/mingw-w64-gcc/) -, then I guess the lowest GCC version to compile this project would be  GCC 5.2 (July 16, 2015) or possibly a bit higher.  

So on my Arch system I did..

sudo pacman -U https://archive.archlinux.org/packages/m/mingw-w64-gcc/mingw-w64-gcc-5.2.0-3-x86_64.pkg.tar.xz

Which installed the version 5.2.0 and I could build the project fine (didn't test the DLLs with a game though).

Tried again with 5.1.0 and 4.9.2, AND 4.9.1 which OP used, and all worked(??).

Is there some other library being used?

```
c0rn3j@c0rn3jDesktop  ~/dumbxinputemu   master  /usr/bin/i686-w64-mingw32-gcc --version
i686-w64-mingw32-gcc (GCC) 4.9.1
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
